src/main.ts

```diff
@@ -182,6 +192,108 @@ class NixInstallerAction {
     actions_core.endGroup();
   }
 
+  // Detect if we are running under `act` or some other system which is not using docker-in-docker,
+  // and instead using a mounted docker socket.
+  // In the case of the socket mount solution, the shim will cause issues since the given mount paths will
+  // equate to mount paths on the host, not mount paths to the docker container in question.
+  async detectDockerWithMountedDockerSocket(): Promise<boolean> {
+    let cgroups_buffer;
+    try {
+      // If we are inside a docker container, the last line of `/proc/self/cgroup` should be
+      // 0::/docker/$SOME_ID
+      //
+      // If we are not, the line will likely be `0::/`
+      cgroups_buffer = await readFile("/proc/self/cgroup", {
+        encoding: "utf-8",
+      });
+    } catch (e) {
+      actions_core.debug(
+        `Did not detect \`/proc/self/cgroup\` existence, bailing on docker container ID detection:\n${e}`,
+      );
+      return false;
+    }
+
+    const cgroups = cgroups_buffer.trim().split("\n");
+    const last_cgroup = cgroups[cgroups.length - 1];
```

Comment:
Would it be worth it to check _every_ cgroup line to see if there is `/docker/` in it, or if it's in an earlier cgroup, is that a weird configuration we wouldn't want to support anyways?

Reply:
I don't have access to every system and setup, only a few, but I think in this case we're only looking for the `0::` prefixed line which is always last.